### PR TITLE
feature/66/Streamlined Digital Resource Management

### LINKED ISSUE #66: feature/Presentation Layer Update Implementation of New Enhancements Digital Resorce
#### 💡 *Description* 💡

This task involves updating the presentation layer of the application to incorporate new enhancements aimed at improving user experience and functionality. The enhancements may include but are not limited to UI redesign, performance optimizations, and the addition of new features.

---

### 🛠️ **Implementation** 🛠️

The implementation process will include:

- Reviewing current presentation layer architecture and identifying areas for enhancement.
- Implementing UI changes and improvements according to design specifications.
- Integrating new features seamlessly into the existing presentation layer.
- Conducting thorough testing to ensure compatibility and functionality across different devices and browsers.

---

### PR DESCRIPTION
**Pull request by BorjaJG**
## 📝 Brief description of the associated ticket to this PR  
The performance issue in the user search function, causing significant delays in the homepage loading, has been addressed.

## 👩‍💻 Summary of the changes introduced  
The user search logic has been refactored to enhance efficiency and reduce waiting times. Additional indexes have been implemented in the database, and queries have been optimized to expedite the search process.

## 📸 Screenshot or Video  
N/A

## ✅ Checklist  
- [x] A descriptive title has been added to the PR.
- [x] I have assigned myself as the author.
- [x] The project compiles and runs correctly.
- [x] The code has been tested with all possible options.
- [x] The code has been formatted.
- [x] Test code has been removed.
- [x] Unit tests have been added.

## ✋ Additional Notes (Disclaimer)  
Please review the integration with the existing pagination function, as it could impact performance.

### CODE REVIEW

**Comment by BorjaJG** [comment]
🚀 **Introduction**

I embarked on refining the digital resource management functionality in my project, opting to centralize most operations within the `DigitalResource` entity. Recognizing the intrinsic nature of digital resources as stand-alone entities, irrespective of their format, such as books or movies, I found it redundant to segregate operations based on type. This architectural refinement not only eradicated redundancy but also bolstered coherence and comprehensibility.

🔧 **Implementation**

Implementation entailed migrating the bulk of resource management logic to the `DigitalResource` presentation layer. Here, pivotal actions like addition, modification, and deletion of resources were consolidated. Conversely, class-specific functionalities, such as listing and retrieval, were retained in corresponding entities like ebooks and movies, enabling users to access supplementary data not directly affiliated with the `DigitalResource` entity.

💡 **Benefits**

These enhancements not only streamline code complexity but also fortify scalability and maintainability. Adhering to an entity-centric approach with `DigitalResource` furnishes a robust framework for seamlessly integrating additional functionalities and resource types in the future, without necessitating substantial architectural overhauls. In essence, these refinements epitomize my enduring commitment to optimizing efficiency and lucidity in software development.
.

File: src/main/java/com/iesam/digitalibrary/digitalresources/ebook/domain/Ebook.java
Comment (AlejandrodePablo) [inline src/main/java/com/iesam/digitalibrary/digitalresources/ebook/domain/Ebook.java:1]: Having the methods declared as final it is not necessary that you put the get methods, calling them with the following format (model.attribute) does the same job

File: src/main/java/com/iesam/digitalibrary/digitalresources/movie/domain/Movie.java
Comment (AlejandrodePablo) [inline src/main/java/com/iesam/digitalibrary/digitalresources/movie/domain/Movie.java:1]: Same as the e-book class.

**Review by AlejandrodePablo — APPROVED** [review]
Review the comments, but nice job! 🍻

**Pull request merged**